pkg/ruler/ruler.go

```diff
@@ -346,15 +350,31 @@ func (r *Ruler) starting(ctx context.Context) error {
 		return errors.Wrap(err, "unable to start ruler subservices")
 	}
 
-	// Wait until the ring client detected this instance in the ACTIVE state to
-	// make sure that when we'll run the initial sync we already know  the tokens
-	// assigned to this instance.
+	// Sync the rule when the ruler JOINING the ring,
+	// Activate the rule evaluation after the ruler is ACTIVE in the ring.
+	// This is to make sure that the ruler is ready to evaluate alert rules once it is ACTIVE in the ring.
```

Comment:
```suggestion
	// This is to make sure that the ruler is ready to evaluate rules immediately after it is ACTIVE in the ring.
```

this doesn't apply to alert rules only